src/run/lib/Src_file.ml

```diff
@@ -111,8 +111,10 @@ let get_region x start end_ =
   else if first_row < last_row then
     let buf = Buffer.create 100 in
     safe_add_rest_of_line buf x first_row start.column;
+    Buffer.add_string buf "\n";
     for row = first_row + 1 to last_row - 1 do
-      safe_add_whole_line buf x row
+      safe_add_whole_line buf x row;
+      Buffer.add_string buf "\n";
```

Comment:
Maybe we shouldn't add a newline here if we've reached the end of the region. I'm not sure what the consequences are. This code is not obvious so we should have tests. This would make the expectations obvious.

Reply:
Let me see if can add something quick.

Reply:
I added unit tests and replaced the implementation of `load_file` by a correct one (doesn't swallow the trailing newline.

One of the new tests is failing. It has to do with how `get_region` deals with the end of the file. It may be an old bug and may not have a big impact at all but it should be fixed.

Reply:
ok, I changed the representation of the lines so that they include the line terminator. I think it's the correct way of proceeding. It's now possible to extract a region (token) ending with `\n` without any special logic.

Reply:
I leave the PR open in case you think of something but otherwise, you can merge anytime.